actions/push-to-gar-docker/action.yaml

```
@@ -25,6 +25,11 @@ inputs:
     description: |
       List of arguments necessary for the Docker image to be built.
     default: ""
+  push:
+    description: |
+      Whether to push the image to the registry.
+    required: false
+    default: "true"
```

will this work?

```suggestion
    default: ${{ github.event_name == 'push' }}
```

this is a break of backwards compatibility otherwise

---

so I pointed [this build](https://github.com/grafana/oncall-private/actions/runs/9367659047/job/25787982173#step:3:12) to commit 9cc5c0d2328834aa95874592e4efba3d0c91f17e, it looks like this works (this was a `push` event, `push` without being specified defaulted to `true`).

On [this build](https://github.com/grafana/oncall-private/actions/runs/9367858725/job/25788469616?pr=2739#step:3:8), a `pull_request` event, I specified `push: true` and it worked as expected